prov/efa/test/efa_unit_test_send.c

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) Amazon.com, Inc. or its affiliates. All rights reserved.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include "efa_unit_tests.h"
+#include "ofi_util.h"
+#include "efa_rdm_ep.h"
+
+void test_efa_rdm_msg_send_to_local_peer_with_null_desc(struct efa_resource **state)
+{
+        struct efa_resource *resource = *state;
+        char buf[10];
+        int i;
+        struct iovec iov;
+        struct efa_ep_addr raw_addr;
+	size_t raw_addr_len = sizeof(raw_addr);
+        fi_addr_t addr;
+        int ret;
+        struct fi_msg msg = {0};
+        struct fi_msg_tagged tmsg = {0};
+
+        efa_unit_test_resource_construct(resource, FI_EP_RDM);
+
+        ret = fi_getname(&resource->ep->fid, &raw_addr, &raw_addr_len);
+	assert_int_equal(ret, 0);
+
+	raw_addr.qpn = 1;
+	raw_addr.qkey = 0x1234;
+	ret = fi_av_insert(resource->av, &raw_addr, 1, &addr, 0 /* flags */, NULL /* context */);
+	assert_int_equal(ret, 1);
+
+        for (i = 0; i < 10; i++)
+                buf[i] = 'a' + i;
+
+        iov.iov_base = buf;
+        iov.iov_len = 10;
+
+        msg.msg_iov = &iov;
```

Comment:
nit: can move the msg and tmsg construction into a small function and reuse it. Function can be in `efa_unit_test_common.c`

Reply:
Done